p2p/subscriber.go

```diff
@@ -164,19 +159,9 @@ func (s *Subscriber[H]) verifyMessage(
 		}
 	}()
 
-	hdr := header.New[H]()
-	err := hdr.UnmarshalBinary(msg.Data)
-	if err != nil {
-		log.Errorw("unmarshalling header",
-			"from", p.ShortString(),
-			"err", err)
-		s.metrics.reject(ctx)
-		return pubsub.ValidationReject
-	}
-	// ensure header validity
-	err = hdr.Validate()
+	hdr, err := s.extractHeader(msg)
 	if err != nil {
-		log.Errorw("invalid header",
+		log.Errorw("extracting header",
```

Comment:
Let's keep it invalid. Worried that "extracting" might be confusing as it unclear without code context what exactly that means.

Reply:
Fair enough. Fixed.